File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/dayview/IDayViewContract.java
package com.wiseassblog.samsaradayplanner.ui.dayview;

import com.wiseassblog.samsaradayplanner.domain.Day;
import com.wiseassblog.samsaradayplanner.domain.Tasks;

public interface IDayViewContract {
    interface View {
        void setData(Day day, Tasks tasks);
        void navigateToHourView(int hourInteger);
        void navigateToTasksView();
        void showMessage(String message);
        void restartFeature();
    }

    interface ViewModel {
        void setDay(Day day);
        Day getDay();
        void setTasks(Tasks tasks);
        Tasks getTasks();
    }

}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/dayview/IDayViewContract.kt
package com.wiseassblog.samsaradayplanner.ui.dayview

import com.wiseassblog.samsaradayplanner.domain.Day
import com.wiseassblog.samsaradayplanner.domain.Tasks

interface IDayViewContract {
    interface View {
        fun setData(day: Day, tasks: Tasks)
        fun navigateToHourView(hourInteger: Int)
        fun navigateToTasksView()
        fun showMessage(message: String)
        fun restartFeature()
    }

    interface ViewModel {
        var day: Day?
        var tasks: Tasks?
    }
}